main/src/cgeo/geocaching/utils/FilterUtils.java

```diff
@@ -0,0 +1,107 @@
+package cgeo.geocaching.utils;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.activity.FilteredActivity;
+import cgeo.geocaching.filters.core.GeocacheFilter;
+import cgeo.geocaching.filters.core.GeocacheFilterContext;
+import cgeo.geocaching.filters.gui.GeocacheFilterActivity;
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.ui.TextParam;
+import cgeo.geocaching.ui.dialog.SimpleDialog;
+
+import android.app.Activity;
+import android.os.Handler;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.TextView;
+import static android.view.View.GONE;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class FilterUtils {
+
+    private FilterUtils() {
+        // should not be instantiated
+    }
+
+    public static void openFilterActivity(final Activity activity, final GeocacheFilterContext filterContext, final Collection<Geocache> filteredList) {
+
+        GeocacheFilterActivity.selectFilter(
+            activity,
+            filterContext,
+            filteredList, true);
+    }
+
+    public static boolean openFilterList(final Activity activity, final GeocacheFilterContext filterContext) {
+        final List<GeocacheFilter> filters = new ArrayList<>(GeocacheFilter.Storage.getStoredFilters());
+
+        if (filters.isEmpty()) {
+            return false;
+        } else {
+            final boolean isFilterActive = filterContext.get().isFiltering();
+            final FilteredActivity filteredActivity = (FilteredActivity) activity;
```

Comment:
This will generate a classcastexception in case activity is not a filteredactivity. Can you make the parameter of type "FilteredActivity"? (this is a minor thing)

Reply:
Unfortunately `FilteredActivity` is not an `Activity` and `SimpleDialog`needs an activity. Only thing I can probably do is to pass it as an additional parameter.

Or is there a possibility to check whether its of type of `FilteredActivity` ? I thought the cast would leed to a null-object...

Reply:
You can check with `if (activity instanceof FilteredActivity) { ... }`

Reply:
done with commit abf0f09

Reply:
Just for the record: if you want to force a method parameter to extend two different classes/interfaces then you can do something like this (using generics):

```
    public static <T extends Activity & FilteredActivity> void doSomething(final T param) {
        //do stuff here...
    }
```

But the solution proposed by @moving-bits is totally fine too of course.

Reply:
But I like your approach more 😉 I will change it

Reply:
> Just for the record: if you want to force a method parameter to extend two different classes/interfaces then you can do something like this (using generics):

Ah, didn't know that. Way more elegant if the checks can be done at compile-time.